pkg/util/labels.go

```diff
@@ -60,27 +60,50 @@ func addDeploymentConfigNestedLabels(obj *deployapi.DeploymentConfig, labels lab
 	return nil
 }
 
+// addUnstructuredNestedLabels adds new label(s) to a nested labels of a single unstructured object
+func addUnstructuredNestedLabels(obj *runtime.Unstructured, labels labels.Set) error {
+	if template, ok := obj.Object["template"]; ok {
```

Comment:
I'm not confident about returning an error here. Different than the other `add*Labels` functions this one receives an object that's not structured (yet) and can be of multiple popential types (`ImageStream`, `DeploymentConfig`, `Service` and so on). Forcing an expected structure and erroring out if it doesn't fit seems wrong, it's more like "add nested labels to this object *where it applies*". Notice that we still return an error if the structure fits but the merging fails.